app/models/immunisation_import.rb

```diff
@@ -167,12 +200,36 @@ def delivery_method
       end
     end
 
+    def recorded_at
+      Time.zone.now
+    end
+
     def organisation_code
-      @data["ORGANISATION_CODE"]
+      @data["ORGANISATION_CODE"]&.strip
     end
 
-    def recorded_at
-      Time.zone.now
+    def patient_first_name
+      @data["PERSON_FORENAME"]&.strip
+    end
+
+    def patient_last_name
+      @data["PERSON_SURNAME"]&.strip
+    end
+
+    def patient_date_of_birth
+      @data["PERSON_DOB"]&.strip
+    end
+
+    def patient_nhs_number
+      @data["NHS_NUMBER"]&.strip
+    end
```

Comment:
Can we guarantee that this will contain no superfluous spaces?

I notice [we remove them in the Patient model](https://github.com/nhsuk/manage-vaccinations-in-schools/blob/55e53ac325fc1620dfc4ea1bd531be069f37cd8f/app/models/patient.rb#L75) but that only happens at the point of saving, so here we could end up in a situation where we try and create multiple patients with the same NHS number if one has some spaces in and the other doesn't.

Reply:
Yeah, I thought it would encounter the validation at the point of saving and stop it there.

I've replaced the `.strip` with the gsub call instead 👍